File: 2274-keep-multiplying-found-values-by-two/2274-keep-multiplying-found-values-by-two.java
class Solution {
    public int findFinalValue(int[] nums, int original) {
        
        while(true){
            int prev = original ;
            for(int i = 0 ; i < nums.length ; i++){
                if(nums[i] == original)
                    original *= 2 ;
            }
            if(prev == original)    break ;
        }
        return original ;
    }
}

File: 2274-keep-multiplying-found-values-by-two/2274-keep-multiplying-found-values-by-two.cpp
class Solution {
public:
    int findFinalValue(vector<int>& nums, int original) {
        set<int> st ;
        for(int num : nums)
            st.insert(num) ;
        while(st.contains(original))
            original *= 2 ;
        
        return original ;
    }
};